### PR TITLE
ROS#56 - API design

### PR DESCRIPTION
**Pull request by Glamhoth**

### CODE REVIEW

File: src/api/runtime_api.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/// System runtime API.
+pub trait RuntimeApi: ErrorType {
+    /// Gets current system time timestamp.
+    fn get_system_time(&'static self) -> f64;
+
+    /// Sets system time offset.
+    ///
+    /// * `offset` - Time offset.
+    fn set_system_time_offset(&'static self, offset: f64);
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply src/api/runtime_api.rs:15]: Sounds reasonable, if offset will be kept separately from time counter itself.

File: src/aerugo.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    fn subscribe_tasklet_to_cyclic<T>(
+        &'static self,
+        _tasklet: &TaskHandle<T>,
+        _period: f64,
+    ) -> Result<(), Self::Error> {
+        todo!()
+    }
+
+    fn init_hardware(&self, _init_fn: fn(&Peripherals)) {
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/aerugo.rs:93]: I'm not sure, and it's hard to guess right now, as we don't have any concrete design for the peripheral access yet. This API will for sure undergo some changes during the actual implementation, for now I only wanted to sketch out how user will be able to configure those peripherals. I will change this to `mut` for now, it looks more reasonable.

File: src/api/runtime_api.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/api/runtime_api.rs:15]: Yes, that's the plan for now.

File: src/event/event_handle.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    /// Emits an event that is delayed in time
+    ///
+    /// * `event` - Event to emit.
+    /// * `delay` - Delay amount.
+    ///
+    /// Returns `RuntimeError` in case of an error, `Ok(())` otherwise.
+    #[inline(always)]
+    pub fn emit_delayed(&self, _delay: f64) -> Result<(), RuntimeError> {
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/event/event_handle.rs:30]: I replaced those floats with `Instant` and `Duration` from [fugit](https://docs.rs/fugit/0.3.6/fugit/) crate. It looks okay for now, will be confirmed during implementation.

File: src/aerugo.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    fn subscribe_tasklet_to_conditions<T>(
+        &'static self,
+        _tasklet: &TaskHandle<T>,
+        _conditions: BooleanConditionSet,
+    ) -> Result<(), Self::Error> {
+        todo!()
+    }
+
+    fn subscribe_tasklet_to_cyclic<T>(
```
Comment (Lurkerpas) [inline src/aerugo.rs:87]: cyclic execution? period?

File: src/event/event_handle.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lurkerpas) [reply src/event/event_handle.rs:30]: [accent] Yes, you have replaced the first set of time floats, but what about the second set of time floats? [/accent] like the ones used in emit_delayed?

[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/event/event_handle.rs]: Done

File: src/api/init_api.rs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    /// Subscribes tasklet to the cyclic execution.
+    ///
+    /// * `tasklet` - Handle to the target tasklet.
+    /// * `period` - Time period of the execution.
+    ///
+    /// Returns `Error` in case of an error, `Ok(())` otherwise.
+    fn subscribe_tasklet_to_cyclic<T>(
+        &'static self,
+        tasklet: &TaskHandle<T>,
+        period: f64,
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/api/init_api.rs:106]: Done, `f64` is no more.

File: src/aerugo.rs [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        &'static self,
+        _tasklet: &TaskHandle<T>,
+        _period: f64,
```
[the remark this answers is not in the document]
Comment (Glamhoth) [reply src/aerugo.rs:90]: done

**Pull request merged**